test/CheckedC/bounds-decl-checking.c

```diff
@@ -3,14 +3,46 @@
 // mostly unimplemented, we only issue warnings when bounds declarations
 // cannot be provided to hold.
 //
-// RUN: %clang -cc1 -fcheckedc-extension -Wcheck-bounds-decls -verify -verify-ignore-unexpected=note %s
+// RUN: %clang -cc1 -fcheckedc-extension -Wcheck-bounds-decls -verify %s
 
+// Initialization by null - no warning.
 void f1(_Array_ptr<int> p : bounds(p, p + x), int x) {
     _Array_ptr<int> r : bounds(p, p + x) = 0;
-    r = p;
 }
 
-void f2(_Array_ptr<int> p : count(x), int x) {
+// Initialization by expression with syntactically identical
+// normalized bounds - no warning.
+void f2(_Array_ptr<int> p : bounds(p, p + x), int x) {
+  _Array_ptr<int> r : bounds(p, p + x) = 0;
```

Comment:
The comment says "identical normalized bounds" but the initializer is null. Typo, or my misunderstanding?